contracts/mint/src/lib.rs

```diff
@@ -48,7 +49,11 @@ impl Mint {
 
 #[no_mangle]
 unsafe extern "C" fn init() {
-    MINT = Some(Mint::default());
+    let arean_contract: ActorId = msg::load().expect("Can't load id of arena contract");
```

Comment:
arena contract requires address of mint contract. it makes impossible to instantiate mint contract before arena...
my suggestion is to make `arena_contract` field optional and assign arena address via additional action - smth like `MintAction::SetArena`

Reply:
we also should check that the assignment is done by the owner of mint contract